### ISSUE TITLE
Add missing things to Azure pipelines

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#15385]
See https://github.com/sympy/sympy/pull/15297

The following are missing from Azure pipelines

- [ ] PyPy (couldn't figure out how to get it working, waiting for https://github.com/Microsoft/vsts-tasks/issues/8514 which should make it easier)
- [ ] The optional dependency build is currently an allowed failure, and fails due to libgfortran issues. It should (hopefully) start to work once the compiler migration on conda forge is finished. We should also move the 3.6 build to 3.7 when that happens.
- [ ] The Sphinx docs are not built on Azure. We will always need them on Travis because doctr only runs on Travis, so I didn't bother adding another job for it.
- [ ] The rigorous setup.py install test from Travis (TEST_SETUP).
- [ ] The slow tests

**Comment by asmeurer** [comment]
I just switched to oauth via @sympy-bot. Let's see if it still works.

**Comment by sidhantnagpal** [comment]
If it helps, recently [Travis](https://blog.travis-ci.com/2018-10-11-windows-early-release) also added support for Windows.

**Comment by asmeurer** [comment]
It isn't working. @vtbassmatt do you have any ideas what might be going on? I disabled the Apps integration and did an oauth integration via the @sympy-bot use, and now no builds work, even a manually triggered build gives `An error occurred while attempting to resolve a specific version from the source branch 'refs/heads/master'. Expected a GitHub connected service ID or a personal access token with this pipeline`. 

I had a hard time getting the permissions right on Azure for the @sympy-bot user to get it to be able to authorize, so I may have messed up. I do have Azure listed on the https://github.com/settings/applications page for @sympy-bot.

**Comment by asmeurer** [comment]
OK, I will play around with that. 

The reason I am trying to switch to oauth is because the apps integration makes it use the checks tab, which is not useful (see https://github.com/sympy/sympy/pull/15280#issuecomment-426795606). Also partly because of https://github.com/sympy/sympy/pull/15280#issuecomment-427194249.

**Comment by asmeurer** [comment]
OK, I did that and it looks like the bot had to have admin access to add the hook (but it only told me that when I tried to create a new pipeline). I think it is working now. Now it looks like both pipelines are building, so I will delete the new one. Hopefully I can remove admin access from the @sympy-bot user and it will continue to work.

**Comment by asmeurer** [comment]
Why does it work on Travis? Is it because Travis has its own libgfortran which it is linking against?

**Comment by isuruf** [comment]
Yes, in Travis, libgfortran is installed using both conda and apt. https://github.com/sympy/sympy/blob/master/.travis.yml#L34

**Comment by asmeurer** [comment]
Ah that explains it. This is code that is being compiled automatically by SymPy. What is the best way to handle this sort of thing? CC @bjodah

**Comment by isuruf** [comment]
Link libgfortran statically?

**Comment by asmeurer** [comment]
I opened a new issue to track this https://github.com/sympy/sympy/issues/15671

**Comment by bjodah** [comment]
Whenever I compile things inside a conda environment I always set `$CPATH`, `$LIBRARY_PATH` and `$LD_LIBRARY_PATH` to `$CONDA_PREFIX/include` and `$CONDA_PREFIX/lib` respectively.

**Comment by asmeurer** [comment]
https://groups.google.com/d/msg/sympy/jyYrYz8ys9o/p7IY4MCSAwAJ

**Comment by asmeurer** [comment]
Thank you. I cannot complain about the support from the Azure team. 

The biggest feature request I would have would be to automatically give people who have push access on GitHub the appropriate permissions on Azure pipelines.

**Comment by vtbassmatt** [comment]
I agree, that would go a long way. Thanks again for the feedback.
